Misc/NEWS.d/next/Core_and_Builtins/2024-10-22-04-18-53.gh-issue-125498.cFjPIn.rst

```diff
@@ -0,0 +1,6 @@
+The JIT has been updated to leverage LLVM 19’s preserve_none attribute, which
+adds native support for the Glasgow Haskell Compiler Calling Convention (ghccc)
+for both x86-64 and AArch64 targets. This removes the need to manually patch
+the calling convention in LLVM IR, simplifying the JIT compilation process.
+Additionally, this update enables the optimization across all platforms, except
+for i686-pc-windows-msvc where the attribute is not supported.
```

Comment:
Hm, I'd reword this. The new calling convention is actually different from the GHC calling convention. And I don't think we need to mention 32-bit Windows.

How about this?

```suggestion
The JIT has been updated to leverage Clang 19’s new ``preserve_none`` attribute,
which supports more platforms and is more useful than LLVM's existing `ghccc`
calling convention. This also removes the need to manually patch the calling
convention in LLVM IR, simplifying the JIT compilation process.
```

Reply:
Ah yeah that's fair - it's kind of like ghccc but not quite (I was rifting on some of the language used in https://discourse.llvm.org/t/rfc-exposing-ghccc-calling-convention-as-preserve-none-to-clang/74233)

Reply:
Yeah, they ended up changing it quite a bit in the end to use a more sane set of registers, haha.